modules/renter/hostdb/hostentry_test.go

```diff
@@ -0,0 +1,150 @@
+package hostdb
+
+import (
+	"testing"
+	"time"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// TestInsertHost tests the insertHost method, which also depends on the
+// scanHostEntry method.
+func TestInsertHost(t *testing.T) {
+	hdb := bareHostDB()
+
+	// invalid host should not be scanned
+	hdb.insertHost(modules.HostSettings{NetAddress: "foo"})
+	select {
+	case <-hdb.scanPool:
+		t.Error("invalid host was added to scan pool")
+	case <-time.After(10 * time.Millisecond):
+	}
+
+	// valid host should be scanned
+	hdb.insertHost(modules.HostSettings{NetAddress: "foo:1234"})
+	select {
+	case <-hdb.scanPool:
+	case <-time.After(10 * time.Millisecond):
+		t.Error("host was not scanned")
+	}
+
+	// duplicate host should not be scanned
+	hdb.insertHost(modules.HostSettings{NetAddress: "foo:1234"})
+	select {
+	case <-hdb.scanPool:
+		t.Error("duplicate host was added to scan pool")
+	case <-time.After(10 * time.Millisecond):
+	}
+}
+
+// TestActiveHosts tests the ActiveHosts method.
+func TestActiveHosts(t *testing.T) {
+	hdb := bareHostDB()
+
+	// empty
+	if hosts := hdb.ActiveHosts(); len(hosts) != 0 {
+		t.Errorf("wrong number of hosts: expected %v, got %v", 0, len(hosts))
+	}
+
+	// with one host
+	h1 := new(hostEntry)
+	h1.NetAddress = "foo"
+	hdb.activeHosts = map[modules.NetAddress]*hostNode{
+		h1.NetAddress: &hostNode{hostEntry: h1},
+	}
```

Comment:
I don't think that this is sufficient. If you want to test activeHosts in this way, you should write an `insertActiveHost` function that does the insertion for you. Then, any time an active host is being inserted within the codebase, it should call `insertActiveHost`, and you should have unit tests on the other functions which increase the active host count to make sure that they are increasing the active host count by calling `insertActiveHost`.

So, now I do see the usefulness of the type of test I was complaining about above. It's a good way to make sure that functions aren't taking shortcuts which may increase coupling between different parts of the codebase.